qualtran/serialization/resolver_dict.py

```diff
@@ -0,0 +1,311 @@
+    "qualtran.bloqs.rotations.phasing_via_cost_function.PhasingViaCostFunction": qualtran.bloqs.rotations.phasing_via_cost_function.PhasingViaCostFunction,
+    "qualtran.bloqs.rotations.programmable_rotation_gate_array.ProgrammableRotationGateArray": qualtran.bloqs.rotations.programmable_rotation_gate_array.ProgrammableRotationGateArray,
+    "qualtran.bloqs.rotations.programmable_rotation_gate_array.ProgrammableRotationGateArrayBase": qualtran.bloqs.rotations.programmable_rotation_gate_array.ProgrammableRotationGateArrayBase,
+    "qualtran.bloqs.rotations.quantum_variable_rotation.QvrInterface": qualtran.bloqs.rotations.quantum_variable_rotation.QvrInterface,
+    "qualtran.bloqs.rotations.quantum_variable_rotation.QvrPhaseGradient": qualtran.bloqs.rotations.quantum_variable_rotation.QvrPhaseGradient,
+    "qualtran.bloqs.rotations.quantum_variable_rotation.QvrZPow": qualtran.bloqs.rotations.quantum_variable_rotation.QvrZPow,
+    "qualtran.bloqs.select_and_prepare.PrepareOracle": qualtran.bloqs.select_and_prepare.PrepareOracle,
+    "qualtran.bloqs.select_and_prepare.SelectOracle": qualtran.bloqs.select_and_prepare.SelectOracle,
+    "qualtran.bloqs.state_preparation.prepare_uniform_superposition.PrepareUniformSuperposition": qualtran.bloqs.state_preparation.prepare_uniform_superposition.PrepareUniformSuperposition,
+    "qualtran.bloqs.state_preparation.state_preparation_alias_sampling.StatePreparationAliasSampling": qualtran.bloqs.state_preparation.state_preparation_alias_sampling.StatePreparationAliasSampling,
+    "qualtran.bloqs.state_preparation.state_preparation_via_rotation.PRGAViaPhaseGradient": qualtran.bloqs.state_preparation.state_preparation_via_rotation.PRGAViaPhaseGradient,
+    "qualtran.bloqs.state_preparation.state_preparation_via_rotation.StatePreparationViaRotations": qualtran.bloqs.state_preparation.state_preparation_via_rotation.StatePreparationViaRotations,
+    "qualtran.bloqs.swap_network.cswap_approx.CSwapApprox": qualtran.bloqs.swap_network.cswap_approx.CSwapApprox,
+    "qualtran.bloqs.swap_network.multiplexed_cswap.MultiplexedCSwap": qualtran.bloqs.swap_network.multiplexed_cswap.MultiplexedCSwap,
+    "qualtran.bloqs.swap_network.swap_with_zero.SwapWithZero": qualtran.bloqs.swap_network.swap_with_zero.SwapWithZero,
+    "qualtran.bloqs.util_bloqs.Allocate": qualtran.bloqs.util_bloqs.Allocate,
+    "qualtran.bloqs.util_bloqs.ArbitraryClifford": qualtran.bloqs.util_bloqs.ArbitraryClifford,
+    "qualtran.bloqs.util_bloqs.Cast": qualtran.bloqs.util_bloqs.Cast,
+    "qualtran.bloqs.util_bloqs.Free": qualtran.bloqs.util_bloqs.Free,
+    "qualtran.bloqs.util_bloqs.Join": qualtran.bloqs.util_bloqs.Join,
+    "qualtran.bloqs.util_bloqs.Partition": qualtran.bloqs.util_bloqs.Partition,
+    "qualtran.bloqs.util_bloqs.Split": qualtran.bloqs.util_bloqs.Split,
+}
+
+
+def add_to_resolver_dict(*bloqs: Type[Bloq]):
+    RESOLVER_DICT.update({bloq.__module__ + '.' + bloq.__qualname__: bloq for bloq in bloqs})
```

Comment:
Add a fully qualified name of the bloq to the resolver dict, instead of asking users to do this. The goal is to let users update the resolver_dict with custom types they may have defined outside of the library (eg: in a jupyter notebook, in their own project etc.) This is currently invoked from tests as well.